### ISSUE TITLE
Error using tplot_restore

### ISSUE THREAD

**Issue by Kamayanirichhariya** [spedas/pyspedas#682]
Hello,
I am using tplot_restore to restore a tplot file but I am getting shown the following error message. Any help would be appreciated! Thank you!

ValueError                                Traceback (most recent call last)
Cell In[12], line 1
----> 1 pytplot.tplot_restore('C:/Users/Admine/Kamayani_Richhariya/mvn_data_20180731_v2.tplot')

File ~\Kamayani_Richhariya\pyspedas\Lib\site-packages\pytplot\importers\tplot_restore.py:136, in tplot_restore(filename)
    133     if len(temp_v_data.shape) == 2:
    134         temp_v_data = np.transpose(temp_v_data)
--> 136     store_data(data_name, data={'x':temp_x_data, 'y':temp_y_data, 'v':temp_v_data})
    137 else:
    138     #Change from little endian to big endian, since pandas apparently hates little endian
    139     #We might want to move this into the store_data procedure eventually
    140     if (temp_x_data.dtype.byteorder == '>'):

File ~\Kamayani_Richhariya\pyspedas\Lib\site-packages\pytplot\store_data.py:216, in store_data(name, data, delete, newname, attr_dict)
    213 coordinate_list = sorted(list(data.keys()))
    214 dimension_list = [d + '_dim' for d in coordinate_list]
--> 216 temp = xr.DataArray(values, dims=['time']+dimension_list,
    217                         coords={'time': ('time', times)})
    219 if spec_bins_exist:
    220     try:

File ~\Kamayani_Richhariya\pyspedas\Lib\site-packages\xarray\core\dataarray.py:450, in DataArray.__init__(self, data, coords, dims, name, attrs, indexes, fastpath)
    448 data = _check_data_shape(data, coords, dims)
    449 data = as_compatible_data(data)
--> 450 coords, dims = _infer_coords_and_dims(data.shape, coords, dims)
    451 variable = Variable(dims, data, attrs, fastpath=True)
    453 if not isinstance(coords, Coordinates):

File ~\Kamayani_Richhariya\pyspedas\Lib\site-packages\xarray\core\dataarray.py:173, in _infer_coords_and_dims(shape, coords, dims)
    171     dims = tuple(dims)
    172 elif len(dims) != len(shape):
--> 173     raise ValueError(
    174         "different number of dimensions on data "
    175         f"and dims: {len(shape)} vs {len(dims)}"
    176     )
    177 else:
    178     for d in dims:

ValueError: different number of dimensions on data and dims: 1 vs 2

**Comment by jameswilburlewis** [comment]
Thanks for letting us know!  Can you attach your tplot file to this issue?  I'd like to open it in IDL to see what's supposed to be stored in it, then reproduce the issue in pyspedas.

**Comment by jameswilburlewis** [comment]
I've finally had a chance to look at the tplot restore file you emailed me.  It looks like there are some minor problems with the variables
'wake', 'twake', and 'mvn_swim_density'.  It seems that each of these variables has scalar values versus time, but there is a 'v' attribute defined (as if they were vectors or spectra).   And at least for 'wake', the 'v' attribute appears to be a one-element array with a string value.   store_data had problems converting these variables, but I was able to ignore the 'v' attribute and successfully create the variables.

Something stranger was going on with the variables 'alt2' and 'stat'.   They seem to be structured differently (somehow) from the other variables in the .tplot file, and I'm not sure how they should be handled.

Were you the one who created the tplot save file?   Would you be able to show the results (contents of 'd' structure) of
 
get_data,'varname',data=d

for each of the variables wake, twake, mvn_swim_density, alt2, and stat?

Here is the list of variables that I was successfully able to read after patching tplot_restore and store_data:

```
0 : alt
1 : sza
2 : sheath
3 : pileup
4 : wake
5 : wind
6 : iono
7 : alt_lab
8 : mvn_sun_bar
9 : twind
10 : tsheath
11 : tpileup
12 : twake
13 : period
14 : palt
15 : plon
16 : plat
17 : psza
18 : lon
19 : lat
20 : orbnum
21 : mvn_B_full_MAVEN_MSO
22 : mvn_B_full_MAVEN_MSO_x
23 : mvn_B_full_MAVEN_MSO_y
24 : mvn_B_full_MAVEN_MSO_z
25 : mvn_B_mso_mag
26 : mvn_mod_bcrust_mso
27 : mvn_mod_bcrust_amp
28 : mvn_B_model_mso
29 : mvn_mod_bcrust_mso_x
30 : mvn_mod_bcrust_mso_y
31 : mvn_mod_bcrust_mso_z
32 : mvn_B_mso_x
33 : mvn_B_mso_y
34 : mvn_B_mso_z
35 : mvn_swim_density
36 : mvn_swim_velocity
37 : mvn_swim_velocity_mso
38 : mvn_swim_pressure
39 : mvn_swim_temperature
40 : mvn_swim_temperature_mso
```

Perhaps that's good enough to keep you going in the Python workflow you're performing?   It would be great if we could figure out the other variables, but if not, I can go ahead and commit these changes to a future pytplot-mpl-temp release so you can give it a try with my tplot_restore and store_data patches in place.

**Comment by jameswilburlewis** [comment]
Fixed in pyspedas 1.6.1 / pytplot-mpl-temp 2.2.45
